main.js

```diff
@@ -100,10 +96,25 @@ function displaySearchResults() {
       const keywordsPara = articleHTML.querySelector('.keywords');
 
       article.keywords.forEach(keyword => {
```

Comment:
the `displaySearchResults` method is becoming a little big and hard to read. usually, I am not a fan of breaking every function in super tiny bits, but I think here we could improve the readability by adding a few functions which group the logic semantically.